foyer/validator.py

```diff
@@ -154,9 +154,13 @@ def validate_smarts(self):
                             None, entry.sourceline)
                         errors.append(undefined)
         raise_collected(errors)
-        if missing_smarts:
+        if missing_smarts and debug:
             warn("The following atom types do not have smarts definitions: {}".format(
                 ', '.join(missing_smarts)), ValidationWarning)
+       	if missing_smarts and not debug:
```

Comment:
I think this is tabbed over too far - will never get called inside the outer `if` block